sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/dialect/api/Dialect.kt

```diff
@@ -0,0 +1,21 @@
+package app.cash.sqldelight.core.dialect.api
+
+import com.squareup.kotlinpoet.ClassName
+
+interface Dialect {
```

Comment:
Not super-fond of this name, as it will eventually bind to a driver implementation for a dialect, which could, in theory, be customized by an end-user (based on some GH issues I've read). Anyway, suggestions welcome!

Reply:
lets go with `SqlDelightDialect`